src/js/chat.js

```diff
@@ -418,143 +437,219 @@ $( document ).ready(function() {
+        console.log("sock closed by an error");
+        onDisconnect();
+    }
+    function onDisconnect() {
+        connected = false;
+        $("#reconnect").removeClass("active");
+        $("#global-chat-messages").append($("chat-loading").detach());
+        $("#chat-loading").show();
+        if (failedAttempts == 0) {
+            $("#chat-loading > #connecting").show();
+            failedAttempts++;
+            initSock();
+        }
+        else {
+            currentTimer = disconnectionTimers[Math.min(failedAttempts - 1, 3)];
+            $("#chat-loading > #connecting").hide();
+            $("#chat-loading > #failed").show();
+            $("#chat-loading > #failed > #timer").text(currentTimer);
+            $("#reconnect").show();
+            $('#chat-input').hide();
+            failedAttempts++;
+            clearInterval(timerInterval);
+            timerInterval = setInterval(updateTimer, 1000);
+        }
+    };
+    function updateTimer() {
+        currentTimer--;
+        $("#chat-loading > #failed > #timer").text(currentTimer);
+        if (currentTimer <= 0) {
+            initSock();
+        }
+    }
+
+
+    sock.onmessage = function (e) {
+        var commands = parseCommands(e.data);
+        for (var i = 0; i < commands.length; i++) {
+            var cmd = commands[i];
+            switch (cmd.command) {
+                case "PING":
+                    sock.send("PONG :0.0.0.0\r\n");
+                    break;
+                case "433":
+                    // Nick already used, try with fallback
+                    var nickNum = parseInt(NICK.match(/\^(\d+)/)[1]) + 1;
+                    NICK = NICK.replace(/\^(\d+)/, "^" + nickNum);
+                    sock.send("NICK " + NICK + "\r\n");
+                    sock.send("USER " + "anon" + " 0 * :" + USERNAME + "\r\n");
+                    break;
+                case "001":
+                    // Initial info
+                    console.log("Authenticated");
+                    sock.send("JOIN #" + CHANNEL + "\r\n");
+                    break;
+                case "JOIN":
+                    failedAttempts = 0;
+
+                    //$("#chat-content").css("background-color", "rgba(0,0,0,0)");
+                    var nick = cmd.origin.split("!")[0];
+                    if (nick == NICK) {
+                        console.log("Joined " + cmd.params[0]);
+                        console.log("Loading history...");
+                        $.get("http://irc.eternagame.org:8082/history.html", function (data) {
+                            console.log("History recieved");
+                            var messages = data.trim().split("\n");
+                            var firstNewMessage = 0;
+                            var j;
+                            for (j = 0; j < messages.length; j++)
+                                if (postedMessages.indexOf(messages[j].trim()) == -1) {
+                                    break;
```

Comment:
`break` is short, ditch the curly braces and put `break` on the same line as the if statement

Reply:
Could you move `break` to be on the same line as the if?